src/game.cpp

```diff
@@ -3713,8 +3713,13 @@ void Game::processPlayerInteraction(f32 dtime, bool show_hud, bool show_debug)
 	} else if (pointed.type == POINTEDTHING_NODE) {
 		ToolCapabilities playeritem_toolcap =
 				playeritem.getToolCapabilities(itemdef_manager);
-		if (playeritem.name.empty() && hand_def.tool_capabilities != NULL) {
-			playeritem_toolcap = *hand_def.tool_capabilities;
+		if (playeritem.name.empty()) {
+			const ToolCapabilities *handToolcap = hlist
+				? &hlist->getItem(0).getToolCapabilities(itemdef_manager)
+				: itemdef_manager->get("").tool_capabilities;
+
+			if (handToolcap != nullptr)
```

Comment:
`if (handToolcap)` checks for NULL and nullptr: reduces the problem sources which could follow out of this.
`getToolCapabilities` uses NULL btw.

Reply:
@SmallJoker NULL (aka 0) when cast to a pointer is the same as nullptr in C++.

Reply:
See also http://www.stroustrup.com/C++11FAQ.html#nullptr (first example)